Create Regex benchmarks

GregoryTravis opened the pull request:
Includes benchmarks verifying regex and replacer compilation memoization.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.
- [x] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

GregoryTravis committed:
one

GregoryTravis committed:
more benchmarks

GregoryTravis committed:
Merge branch 'develop' into wip/gmt/6200-regex-benches

GregoryTravis committed:
regex + replacer cache benchmarks

GregoryTravis committed:
precompile group_reference_regex

GregoryTravis committed:
real counts

GregoryTravis committed:
options

GregoryTravis committed:
Merge branch 'develop' into wip/gmt/6200-regex-benches

GregoryTravis committed:
add to main

GregoryTravis committed:
Merge branch 'develop' into wip/gmt/6200-regex-benches

GregoryTravis committed:
Merge branch 'develop' into wip/gmt/6200-regex-benches

radeusgd commented on distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex/Internal/Replacer.enso:
```diff
@@ -95,7 +95,7 @@ build_replacement_vector_cached replacement_string pattern =
 build_replacement_vector : Text -> Regex -> Vector Replacement ! No_Such_Group
 build_replacement_vector replacement_string pattern =
-    replacement_pattern = Regex.compile group_reference_regex
+    replacement_pattern = group_reference_regex
```
I'm wondering if this change changes anything?

The module-level definitions are always a 'function', meaning that `group_reference_regex` unfortunately is not a constant that is computed once, but is a 0-argument function that is executed every time the method is called. So we are still calling `Regex.compile` every time.

I still think this change is OK to keep and I prefer to have the `compile` call as part of the definition - if only for the reason that _if_ at some point we get an optimization that will make such constants computed once, this code will benefit from that. But I wanted to point out that in the current state, there is no practical change here (as far as I understand).

GregoryTravis replied:
Yes, I just changed it since nothing needed to use the raw string directly.

radeusgd replied:
Yeah, I do agree that having this typed to `Regex` is nicer than a `Text` constant 👍 Just wanted to share the context about such module "constants" as it's not obvious.

radeusgd commented on test/Benchmarks/src/Text/Regex_Bench.enso:
```diff
@@ -0,0 +1,84 @@
+    (faker.integer 0 100).to_text + (faker.alpha 8) + (faker.integer 0 100).to_text
+
+# "ab12cdefg634hiab12cdefg634hi"
+build_double_two_nums faker =
```
Should this be called `build_four_nums`? 😅

GregoryTravis replied:
Yes "four" is a better way of writing "double_two", changed. 😃

GregoryTravis committed:
Merge branch 'develop' into wip/gmt/6200-regex-benches

GregoryTravis committed:
four_nums

mergify[bot] committed:
Merge branch 'develop' into wip/gmt/6200-regex-benches

mergify[bot] committed:
Merge branch 'develop' into wip/gmt/6200-regex-benches

Commit:
Convert `Array_Like_Helpers.map` to a builtin to reduce stack size (#11363)

The ultimate goal is to reduce the method calls necessary for `Vector.map`.

# Important Notes
- I managed to reduce the number of Java stack frames needed for each `Vector.map` call from **150** to **22** (See https://github.com/enso-org/enso/pull/11363#issuecomment-2432996902)
- Introduced `Stack_Size_Spec` regression test that will ensure that Java stack frames needed for `Vector.map` method call does not exceed **40**.

kazcw committed:
New markdown editor (#11469)

Implements #11240.

https://github.com/user-attachments/assets/4d2f8021-3e0f-4d39-95df-bcd72bf7545b

# Important Notes
- Fix a Yjs document corruption bug caused by `DeepReadonly` being replaced by a stub; introduce a `DeepReadonly` implementation without Vue dependency.
- Fix right panel sizing when code editor is open.
- Fix right panel slide-in animation.
- `Ast.Function` renamed to `Ast.FunctionDef`.
